engine/resources/mount.go

```diff
@@ -0,0 +1,712 @@
+	// fstabPath is the path to the fstab file which defines mounts.
+	fstabPath = "/etc/fstab"
+	// fstabUmask is the umask (permissions) used to edit /etc/fstab.
+	fstabUmask = 0644
+
+	// getStatus64 is an ioctl command to get the status of file backed
+	// loopback devices (i.e. iso file mounts.)
+	getStatus64 = 0x4C05
+	// loopFileUmask is the umask (permissions) used to read the loop file.
+	loopFileUmask = 0660
+
+	// devDisk is the path where disks and partitions can be found, organized
+	// by uuid/label/path.
+	devDisk = "/dev/disk/"
+	// diskByUUID is the location of symlinks for devices by UUID.
+	diskByUUID = devDisk + "by-uuid/"
+	// diskByLabel is the location of symlinks for devices by label.
+	diskByLabel = devDisk + "by-label/"
+	// diskByUUID is the location of symlinks for partitions by UUID.
+	diskByPartUUID = devDisk + "by-partuuid/"
+	// diskByLabel is the location of symlinks for partitions by label.
+	diskByPartLabel = devDisk + "by-partlabel/"
+
+	// dbusStoragedPath is the dbus path that watch eavesdrops on for signals.
+	dbusStoragedPath = "/org/freedesktop/UDisks2"
+)
+
+// vfsTypes is a list of all valid filesystem types taken from
+// /proc/filesystems.
+var vfsTypes = []string{
+	"sysfs",
+	"rootfs",
+	"ramfs",
+	"bdev",
+	"proc",
+	"cpuset",
+	"cgroup",
+	"cgroup2",
+	"tmpfs",
+	"devtmpfs",
+	"configfs",
+	"debugfs",
+	"tracefs",
+	"securityfs",
+	"sockfs",
+	"dax",
+	"bpf",
+	"pipefs",
+	"hugetlbfs",
+	"devpts",
+	"ext3",
+	"ext2",
+	"ext4",
+	"autofs",
+	"pstore",
+	"efivarfs",
+	"mqueue",
+	"selinuxfs",
+	"vfat",
+	"rpc_pipefs",
+	"fuseblk",
+	"fuse",
+	"fusectl",
+	"iso9660",
+}
+
+// defaultMntOps is a map that sets the default mount options for fstab mounts
+var defaultMntOps = map[string]string{"defaults": ""}
+
+// MountRes is a systemd mount resource that adds/removes entries from
+// /etc/fstab, and makes sure the defined device is mounted or unmounted
+// accordingly.
+type MountRes struct {
+	traits.Base
+
+	init *engine.Init
+
+	// State must be exists ot absent. If absent, remaining fields are ignored.
+	State  string            `yaml:"state"`
+	Device string            `yaml:"device"` // location of the device or image
+	Type   string            `yaml:"type"`   // the type of filesystem
+	MntOps map[string]string `yaml:"mntops"` // mount options
+	Freq   int               `yaml:"freq"`   // dump frequency
+	PassNo int               `yaml:"passno"` // verification order
+
+	mount *fstab.Mount // struct representing the mount
+}
+
+// Default returns some sensible defaults for this resource.
+func (obj *MountRes) Default() engine.Res {
+	return &MountRes{
+		MntOps: defaultMntOps,
+	}
+}
+
+// Validate if the params passed in are valid data.
+func (obj *MountRes) Validate() error {
+	var err error
+
+	// validate state
+	if obj.State != "exists" && obj.State != "absent" {
+		return fmt.Errorf("state must be 'exists', or 'absent'")
+	}
+
+	// validate type
+	var typeOK bool
+	for _, t := range vfsTypes {
```

Comment:
You can simplify this loop by using https://github.com/purpleidea/mgmt/blob/master/util/util.go#L52 and returning right away if not found with a short circuit on != absent.